pkg/glusterfs/controllerserver.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2018 The Gluster CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
We have to carefully assess which parts will be linked with kubernetes code. Unfortunately, this has a political dimension in that the kubernetes project is not very friendly towards GPL. Hence we for instance also licensed the heketi api cli parts as apache and lpgl, so that we gain acceptance for linking into kubernetes proper.

But I agree that we should check the possibility of using GPL for this project as much as possible. I am personally a big fan of GPL and believe that this is the true open source, while apache and bsd style licenses are just going the path of least resistance or a misunderstanding of the real spirit of open source (my personal 2 cents) ;-)